src/main/java/ua/utility/kfsdbupgrade/App.java

```diff
@@ -1950,6 +1950,316 @@ private void createDocumentSearchEntries(Connection conn, Statement stmt) {
+				insertStmt.setString(1, docNbr);
+				insertStmt.setString(2, desc.replace("&", "and"));
+				insertStmt.addBatch();
+				if ((i % 10000) == 0) {
+					insertStmt.executeBatch();
+					LOGGER.info(i + " krew_doc_hdr_ext_t entries inserted");
+				}
+				i++;
+			}
+			// catch any straggler statements
+			insertStmt.executeBatch();
+			LOGGER.info(i + " krew_doc_hdr_ext_t entries inserted TOTAL");
+			conn.commit();
+		} catch (Exception ex) {
+			LOGGER.error(ex);
+
+			if (conn != null) {
+				try {
+					conn.rollback();
+				} catch (Exception ex2) {
+					LOGGER.error(ex);
+				};
+			}
+		} finally {
+			try {
+				closeDbObjects(null, insertStmt, legacyRes);
+			} catch (Exception ex) {
+				LOGGER.error(ex);
+			};
+		}
+	}
+
+	/**
+	 * Execute the following prepared statement:
+	 * <code>insert into krew_doc_hdr_ext_t (doc_hdr_ext_id, doc_hdr_id, key_cd, val) values (to_char(KREW_SRCH_ATTR_S.nextval), ?, 'applicationDocumentStatus', ?)</code>
+	 * with the parameters specified by values returned by
+	 * <code>SELECT P.FDOC_NBR, S.REQS_STAT_DESC FROM PUR_REQS_T P, DEPR_PUR_REQS_STAT_T S WHERE P.DEPR_REQS_STAT_CD = S.REQS_STAT_CD</code>
+	 * . All updates are done in a single transaction; any {@link Exception}s
+	 * encountered will cause the transaction to rollback.
+	 *
+	 * @param conn
+	 *            {@link Connection} to a database
+	 * @param stmt
+	 *            {@link Statement} that is immediately blown away and should be
+	 *            removed as a parameter
+	 */
+	private void addPurchaseRequisitionDocStatus(Connection conn, Statement stmt) {
+		logHeader2("Adding Purchase Requistion document status entries.");
+		PreparedStatement insertStmt = null;
+		ResultSet legacyRes = null;
+		try {
+			conn.setAutoCommit(false);
+			stmt = conn.createStatement();
+			insertStmt = conn.prepareStatement("insert into krew_doc_hdr_ext_t (doc_hdr_ext_id, doc_hdr_id, key_cd, val) values (to_char(KREW_SRCH_ATTR_S.nextval), ?, 'applicationDocumentStatus', ?)");
+			int i = 1;
+			//first, get the list of PO document numbers with the matching textual application document status
+			legacyRes = stmt.executeQuery("SELECT P.FDOC_NBR, S.REQS_STAT_DESC FROM PUR_REQS_T P, DEPR_PUR_REQS_STAT_T S WHERE P.DEPR_REQS_STAT_CD = S.REQS_STAT_CD");
+
+			//then loop through the list of PO document numbers, populating and executing the insert statements
+			while (legacyRes.next()) {
+				String docNbr = legacyRes.getString(1);
+				String desc = legacyRes.getString(2);
+				insertStmt.setString(1, docNbr);
+				insertStmt.setString(2, desc.replace("&", "and"));
+				insertStmt.addBatch();
+				if ((i % 10000) == 0) {
+					insertStmt.executeBatch();
+					LOGGER.info(i + " krew_doc_hdr_ext_t entries inserted");
+				}
+				i++;
+			}
+			// catch any straggler statements
+			insertStmt.executeBatch();
+			LOGGER.info(i + " krew_doc_hdr_ext_t entries inserted TOTAL");
+			conn.commit();
+		} catch (Exception ex) {
```

Comment:
Also the try/catch in each method is redundant if we don't want to roll back for each type of doc. Let the methods throw the exception and catch + rollback in updatePurchasingStatuses